From d2d5399798c218b7e29dcc3b233850f15e4a7451 Mon Sep 17 00:00:00 2001
From: Frank Hunleth <fhunleth@troodon-software.com>
Date: Sat, 5 Oct 2024 23:44:07 -0400
Subject: [PATCH] dwc2: fix compiler error for gadget-only mode

---
 drivers/usb/dwc2/core_intr.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/dwc2/core_intr.c b/drivers/usb/dwc2/core_intr.c
index 31a77d751..96b54459f 100644
--- a/drivers/usb/dwc2/core_intr.c
+++ b/drivers/usb/dwc2/core_intr.c
@@ -406,7 +406,10 @@ void dwc2_wakeup_from_lpm_l1(struct dwc2_hsotg *hsotg, bool remotewakeup)
 		call_gadget(hsotg, resume);
 		/* Change to L0 state */
 		hsotg->lx_state = DWC2_L0;
+#if IS_ENABLED(CONFIG_USB_DWC2_HOST) ||	\
+	IS_ENABLED(CONFIG_USB_DWC2_DUAL_ROLE)
 		hsotg->bus_suspended = false;
+#endif
 fail:		dwc2_gadget_init_lpm(hsotg);
 	} else {
 		/* TODO */
-- 
2.34.1

